client/question/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -567,12 +643,12 @@ describe('FeedbackGeneratorService', function() {
 
       expect(paragraphs.length).toEqual(3);
       expect(paragraphs[0].isTextParagraph()).toBe(true);
-      expect(paragraphs[0].getContent()).toEqual(
-          'It looks like your code has a syntax error. ' +
-          'Try to figure out what the error is.');
-      expect(paragraphs[1].isErrorParagraph()).toBe(true);
+      expect(paragraphs[0].getContent()).toMatch(
+          'Error detected(?: on or near line [0-9]+)*:');
```

Comment:
Probably better to do a toEqual() check, since this is a test. The more explicit the better.

Ditto elsewhere.

Reply:
toEqual() matches a specific string or value. Here, I'm matching a regex. Is the suggestion that I can use toEqual to match on a regex (which I'm not sure would work) or not use regex (but then, we would not be matching the more complete pattern)?

Reply:
Oh -- I think I got thrown off by "not matching the more complete pattern". I was just talking about a equality match to the exact text of the error string that should be shown to the user (including the line number information and everything), since that shouldn't change even if the underlying code changes.

Generally, if you want to test various cases of a pattern, try to write a range of tests that are exact-match checks to individual specific cases for that pattern. That's better than testing against the pattern itself, which is sort of "internal" and more of an implementation detail. It also catches errors where the wrong error message is given for a particular case (but it still happens, by chance, to match the regex pattern).

Reply:
OK. Fixed for the next commit.